cloudbridge/cloud/base/resources.py

```diff
@@ -965,3 +884,7 @@ def wait_till_ready(self, timeout=None, interval=None):
             terminal_states=[GatewayState.ERROR, GatewayState.UNKNOWN],
             timeout=timeout,
             interval=interval)
+
+    def delete(self):
+        return self._provider.networking._gateways.delete(self.network_id,
```

Comment:
This and other places needs a ` # pylint:disable=protected-access` comment to indicate that we are doing this on purpose.